amr-wind/equation_systems/vof/volume_fractions.H

```diff
@@ -1,34 +1,249 @@
+// the gradient is computed with a multiplicative factor of -32:
+//  mm = - 32 * grad (c)
+//----------------------------------------------------------------
+//
+// Known problems: the index (1,1,1), i.e. the central cell in the block, never
+occurs:
+// Therefore an isolated droplet will have a normal with all components to zero.
+// Translated into f90 by Stephane Z.
+*/
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE void youngs_fd_normal(
+    int i,
+    int j,
+    int k,
+    amrex::Array4<amrex::Real const> const& volfrac,
+    amrex::Real& mx,
+    amrex::Real& my,
+    amrex::Real& mz) noexcept
+{
+    amrex::Real mm1, mm2;
+
+    mm1 = volfrac(i - 1, j - 1, k - 1) + volfrac(i - 1, j - 1, k + 1) +
+          volfrac(i - 1, j + 1, k - 1) + volfrac(i - 1, j + 1, k + 1) +
+          2.0 * (volfrac(i - 1, j - 1, k) + volfrac(i - 1, j + 1, k) +
+                 volfrac(i - 1, j, k - 1) + volfrac(i - 1, j, k + 1)) +
+          4.0 * volfrac(i - 1, j, k);
+    mm2 = volfrac(i + 1, j - 1, k - 1) + volfrac(i + 1, j - 1, k + 1) +
+          volfrac(i + 1, j + 1, k - 1) + volfrac(i + 1, j + 1, k + 1) +
+          2.0 * (volfrac(i + 1, j - 1, k) + volfrac(i + 1, j + 1, k) +
+                 volfrac(i + 1, j, k - 1) + volfrac(i + 1, j, k + 1)) +
+          4.0 * volfrac(i + 1, j, k);
+    mx = mm1 - mm2;
+
+    mm1 = volfrac(i - 1, j - 1, k - 1) + volfrac(i - 1, j - 1, k + 1) +
+          volfrac(i + 1, j - 1, k - 1) + volfrac(i + 1, j - 1, k + 1) +
+          2.0 * (volfrac(i - 1, j - 1, k) + volfrac(i + 1, j - 1, k) +
+                 volfrac(i, j - 1, k - 1) + volfrac(i, j - 1, k + 1)) +
+          4.0 * volfrac(i, j - 1, k);
+    mm2 = volfrac(i - 1, j + 1, k - 1) + volfrac(i - 1, j + 1, k + 1) +
+          volfrac(i + 1, j + 1, k - 1) + volfrac(i + 1, j + 1, k + 1) +
+          2.0 * (volfrac(i - 1, j + 1, k) + volfrac(i + 1, j + 1, k) +
+                 volfrac(i, j + 1, k - 1) + volfrac(i, j + 1, k + 1)) +
+          4.0 * volfrac(i, j + 1, k);
+    my = mm1 - mm2;
+
+    mm1 = volfrac(i - 1, j - 1, k - 1) + volfrac(i - 1, j + 1, k - 1) +
+          volfrac(i + 1, j - 1, k - 1) + volfrac(i + 1, j + 1, k - 1) +
+          2.0 * (volfrac(i - 1, j, k - 1) + volfrac(i + 1, j, k - 1) +
+                 volfrac(i, j - 1, k - 1) + volfrac(i, j + 1, k - 1)) +
+          4.0 * volfrac(i, j, k - 1);
+    mm2 = volfrac(i - 1, j - 1, k + 1) + volfrac(i - 1, j + 1, k + 1) +
+          volfrac(i + 1, j - 1, k + 1) + volfrac(i + 1, j + 1, k + 1) +
+          2.0 * (volfrac(i - 1, j, k + 1) + volfrac(i + 1, j, k + 1) +
+                 volfrac(i, j - 1, k + 1) + volfrac(i, j + 1, k + 1)) +
+          4.0 * volfrac(i, j, k + 1);
+    mz = mm1 - mm2;
+}
+
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE void mixed_youngs_central_normal(
+    int i,
+    int j,
+    int k,
+    amrex::Array4<amrex::Real const> const& volfrac,
+    amrex::Real& mx,
+    amrex::Real& my,
+    amrex::Real& mz) noexcept
+{
+    amrex::Real m[AMREX_SPACEDIM + 1][AMREX_SPACEDIM];
```

Comment:
Is this supposed to be a vector? If so 
```suggestion
   vs::Vector mvec[AMREX_SPACEDIM+1];
```

Reply:
it should be a 4X3 array